java/com/google/flatbuffers/FlatBufferBuilder.java

```diff
@@ -242,8 +242,9 @@ static ByteBuffer growByteBuffer(ByteBuffer bb, ByteBufferFactory bb_factory) {
         int new_buf_size = old_buf_size == 0 ? 1 : old_buf_size << 1;
         bb.position(0);
         ByteBuffer nbb = bb_factory.newByteBuffer(new_buf_size);
+        new_buf_size = nbb.capacity(); // TODO - Maybe we get the buffer's `limit()`
```

Comment:
What's with this comment? `limit` is the amount of the buffer that is in-use, so that doesn't sound correct to me? We want an empty buffer, so wouldn't that have a limit of 0?

Reply:
@aardappel Please allow me to clarify myself:

limit acutally indicates the limit accessible by the user (there's variance between read and write access). There's currently no guarantee that the factory would actually provide us with a fresh buffer, and therefor - maybe we shouldn't rely on the capacity and use the limit instead.

**Giving it a bit more thought**, I think that this is actually a _weakness/flaw_ in the current code, and therefor I added the  call to `clear()`, which ensures the buffer is to be treated as _empty_. That way, we don't need to rely on the factory implementation to clear the buffer itself.

We should perhaps add a matching comment in the ByteBufferFactory  itself (something along the lines of "all data currently residing in the buffer may be overwritten").

Let me know what you think.